diff --git a/src/options.h b/src/options.h
index 7bac116..67b2e1d 100644
--- a/src/options.h
+++ b/src/options.h
@@ -52,15 +52,15 @@ namespace zutty
       // option       parseType implValue hardDefault helpDescr
       {"altScroll",   NoArg,    "true",    "false",   "Alternate scroll mode"},
       {"autoCopy",    NoArg,    "true",    "false",   "Sync primary to clipboard"},
-      {"bg",          SepArg,   nullptr,   "#000",    "Background color"},
+      {"bg",          SepArg,   nullptr,   "#ffffea",    "Background color"},
       {"boldColors",  NoArg,    "true",    "true",    "Enable bright for bold"},
       {"border",      SepArg,   nullptr,   "2",       "Border width in pixels"},
       {"cr",          SepArg,   nullptr,   nullptr,   "Cursor color"},
       {"display",     SepArg,   nullptr,   nullptr,   "Display to connect to"},
       {"dwfont",      SepArg,   nullptr,   "18x18ja", "Double-width font to use"},
-      {"fg",          SepArg,   nullptr,   "#fff",    "Foreground color"},
-      {"font",        SepArg,   nullptr,   "9x18",    "Font to use"},
-      {"fontsize",    SepArg,   nullptr,   "16",      "Font size"},
+      {"fg",          SepArg,   nullptr,   "#000",    "Foreground color"},
+      {"font",        SepArg,   nullptr,   "go-mono",    "Font to use"},
+      {"fontsize",    SepArg,   nullptr,   "13",      "Font size"},
       {"fontpath",    SepArg,   nullptr,   fontpath,  "Font search path"},
       {"geometry",    SepArg,   nullptr,   "80x24",   "Terminal size in chars"},
       {"glinfo",      NoArg,    "true",    "false",   "Print OpenGL information"},
