diff --git a/sshd-auth.c b/sshd-auth.c
index 53658cf11..b071c59b1 100644
--- a/sshd-auth.c
+++ b/sshd-auth.c
@@ -464,7 +464,6 @@ main(int ac, char **av)
 	__progname = ssh_get_progname(av[0]);
 
 	/* Save argv. Duplicate so setproctitle emulation doesn't clobber it */
-	saved_argc = ac;
 	saved_argv = xcalloc(ac + 1, sizeof(*saved_argv));
 	for (i = 0; (int)i < ac; i++)
 		saved_argv[i] = xstrdup(av[i]);
